controllers/flinkcluster/flinkcluster_updater.go

```diff
@@ -725,8 +731,7 @@ func (updater *ClusterStatusUpdater) deriveJobStatus() *v1beta1.JobStatus {
 					for _, e := range exceptions.Exceptions {
 						newJob.FailureReasons = append(newJob.FailureReasons, e.Exception)
 					}
-				}
-				if observedSubmitter.log != nil {
+				} else if observedSubmitter.log != nil {
```

Comment:
Here the submitter log was being appended even if there are exceptions. Do you see issues with that?

Reply:
now that I think about it ... getting to long perhaps due to possibly duplicated stuff. So it makes sense this change.

Reply:
Exactly, I was afraid of too much log and it should already be in exceptions